src/main/java/hudson/remoting/Channel.java

```diff
@@ -1080,18 +1079,17 @@ public void terminate(@Nonnull IOException e) {
                     logger.log(Level.WARNING, "Failed to close down the reader side of the transport", x);
                 }
                 try {
-                    synchronized (pendingCalls) {
-                        for (Request<?, ?> req : pendingCalls.values())
-                            req.abort(e);
-                        pendingCalls.clear();
+                    for (Request<?, ?> req : pendingCalls.values()) {
```

Comment:
The iterator provided here by the ConcurrentHashMap will be consistent and coherent over the execution of this for loop. Synchronization on the object, where it isn't done elsewhere, is a broken, mixed threading model.

It's possible that something could be added to the map between the time the iterator is created and the following clear() operation is performed. We could additional logic to correctly deal with this. However, this logic is inside a terminate() method. If we miss aborting a few requests, that shouldn't make much difference as the channel is terminating anyway.

If we do need to deal with that threading inconsistency, we need something different than what is currently released.